### PR TITLE
Implement omrsock_bind and omrsock_listen on Unix

### PR DESCRIPTION
**Pull request by caohaley**
- Implement Bind and Listen functions on unix
- Modified tests to try to bind and listen on opened sockets.

Issue: #4919

### CODE REVIEW

File: port/unix/omrsock.c [hunk 1]
```diff
@@ -352,6 +377,49 @@ omrsock_freeaddrinfo(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t hand
...
+int32_t
+omrsock_sockaddr_init(struct OMRPortLibrary *portLibrary, int32_t family, uint8_t *addrNetworkOrder, uint16_t portNetworkOrder, omrsock_sockaddr_t handle)
+{
+	omr_os_sockaddr_in *sockAddr = (omr_os_sockaddr_in *) &handle->data;
+	memset(handle, 0, sizeof(struct OMRSockAddrStorage));
+	sockAddr->sin_family = get_os_family(family);
+	sockAddr->sin_port = portNetworkOrder;
+	memcpy(&(sockAddr->sin_addr.s_addr), addrNetworkOrder, 4);
+
+	return 0;
```
Comment (rwy0717) [inline port/unix/omrsock.c:389]:
```suggestion
	return 0;
```

File: port/unix/omrsock.c [hunk 2]
```diff
@@ -352,6 +377,49 @@ omrsock_freeaddrinfo(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t hand
...
+			 * Check if it is the INADDR_ANY address. We know the top 4 bytes of sockaddr_6->sin6_addr.s6_addr
+			 * are 0's as we just cleared them, so we use them to do the check. If it isn't, set 16 bits of "1"s
+			 * in front of the IPv4 address.
+			 */
+			if (0 != memcmp(sockAddr6->sin6_addr.s6_addr, addrNetworkOrder, 4)){
+				sockAddr6->sin6_addr.s6_addr[10] = 0xFF;
+				sockAddr6->sin6_addr.s6_addr[11] = 0xFF;
+			}
+	} 
+	else {
```
Comment (rwy0717) [inline port/unix/omrsock.c:413]:
```suggestion
	} else {
```

File: port/unix/omrsock.c [hunk 3]
```diff
@@ -352,6 +377,49 @@ omrsock_freeaddrinfo(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t hand
...
+{
+	omr_os_sockaddr_in6 *sockAddr6 = (omr_os_sockaddr_in6*)&handle->data;
+	memset(handle, 0, sizeof (struct OMRSockAddrStorage));
+
+	if (OS_SOCK_AF_INET == get_os_family(family)) {
+		/* To talk IPv4 on an IPv6 socket, we need to map the IPv4 address to an IPv6 format. */
+			sockAddr6 = (omr_os_sockaddr_in6 *)&handle->data;
+			memset(sockAddr6->sin6_addr.s6_addr, 0,16);
+			memcpy(&(sockAddr6->sin6_addr.s6_addr[12]), addrNetworkOrder, 4);
+			/**
```
Comment (rwy0717) [inline port/unix/omrsock.c:403]:
```suggestion
			/*
```

File: port/unix/omrsock.c [hunk 4]
```diff
@@ -352,6 +377,49 @@ omrsock_freeaddrinfo(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t hand
...
+			 * Check if it is the INADDR_ANY address. We know the top 4 bytes of sockaddr_6->sin6_addr.s6_addr
+			 * are 0's as we just cleared them, so we use them to do the check. If it isn't, set 16 bits of "1"s
+			 * in front of the IPv4 address.
+			 */
+			if (0 != memcmp(sockAddr6->sin6_addr.s6_addr, addrNetworkOrder, 4)){
```
Comment (rwy0717) [inline port/unix/omrsock.c:408]:
```suggestion
			if (0 != memcmp(sockAddr6->sin6_addr.s6_addr, addrNetworkOrder, 4)) {
```

File: fvtest/porttest/omrsockTest.cpp [hunk 5]
```diff
@@ -292,18 +294,89 @@ TEST(PortSockTest, getaddrinfo_and_freeaddrinfo)
...
+ * Test functions involving IPv4 and IPv6 socket addresses.
+ *
+ * After creating the address, their ability to bind and listen will be tested.
+ * 
+ * Address families tested include IPv4 and IPv6.
+ *
+ * @note Errors such as failed function calls, failure to create socket addresses, failure 
+ * to set up bind, listen with the socket address, and so on, will be reported.
+ */
+TEST(PortSockTest, socket_address)
```
[the remark this answers is not in the document]
Comment (caohaley) [reply fvtest/porttest/omrsockTest.cpp:323]: Sounds good! I will fix this.

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
@genie-omr build all

**Pull request merged**